Assets/HoloToolkit/Input/README.md

```diff
@@ -1,6 +1,32 @@
 ## [Input]()
 Scripts that leverage the HoloLens input features namely Gaze, Gesture and Voice.
 
+This contains a fully-featured **input module**, which allows you to handle various types of input and send them to any game object being currently gazed at, or any fallback object. It also includes a **cursor** similar to the HoloLens shell cursor that fully leverages the Unity's animation system.
+
+### [Input Module Design](InputModuleDesign)
+The input module is designed to be extensible: it could support various input mechanisms and various types of gazers.
+
+Each input source (hands, gestures, others) implement a **IInputSource** interface. The interface defines various events that the input sources can trigger. The input sources register themselves with the InputManager, whose role it is to forward input to the appropriate game objects. Input sources can be dynamically enabled / disabled as necessary, and new input sources can be created to support different input devices.
+
+Game objects that want to consume input events can implement one or many **input interfaces**, such as:
+
+- **IFocusable** for focus enter and exit. The focus can be triggered by the user's gaze or any other gaze source.
+- **IHoldHandle** for the Windows hold gesture.
+- **IInputHandler** for source up, down and clicked. The source can be a hand that tapped, a clicker that was pressed, etc.
+- **IManipulationHandler** for the Windows manipulation gesture.
+- **INavigationnHandler** for the Windows navigation gesture.
+- **ISourceStateHandler** for the source detected and source lost events.
```

Comment:
You don't need a SourceMoved event, because at any point in time, you can query the IInputSource (that you receive as part of the event data for any of the events) for its position and/or orientation, if available. HandDraggable gives a good example of that. This allows the event system to stay relatively lightweight in terms of number of events sent.